spharpy/samplings/coordinates.py

```diff
@@ -310,3 +320,31 @@ def n_max(self, value):
             self._n_max = None
         else:
             self._n_max = int(value)
+
+    @property
+    def radius_tolerance(self):
+        """Get or set the radius tolerance in meter."""
+        return self._radius_tolerance
+
+    @radius_tolerance.setter
+    def radius_tolerance(self, value):
+        """Get or set the radius tolerance in meter."""
+        if isinstance(value, (int, float)) and value > 0:
+            self._radius_tolerance = float(value)
+        else:
+            raise ValueError(
+                'The radius tolerance must be a number greater than zero')
+
+    def _set_points(self, x, y, z):
+        # set points first to make sure data is valid
+        super()._set_points(x, y, z)
+
+        # check if all points have the same radius
+        radius = self.radius
+        radius_delta = np.max(np.abs(np.mean(radius) - radius))
+        if radius_delta > self.radius_tolerance:
+            raise ValueError(
+                'All points must have the same radius but the deviation from '
+                f'the mean radius is {radius_delta:.3g} m, which exceeds the'
+                f' tolerance of {self.radius_tolerance:.3g} m.')
```

Comment:
This set's the points (in super()._set_points) even if the ValueError is raised.
That's definitely a bug ;)

Reply:
why is it a bug?

if so, we could change the internal handling in pyfar.Coordinates, to split the current ``_set_points`` into ``check_points``to to all the checked and braodcasting and then the ``_set_points``
(same for _set_weights)
we could do it for 0.8.0, it just changing the backend, so no need to deprecate anyting in pyfar

Reply:
This is how I interpret the code, please correct me if I'm wrong here.
You want to avoid setting points which do not share the same radius, but in the process the points are set anyways. Afterwards an error is raised indicating that the operation failed, however, the internal class values have already been overwritten before the error was raised.

Reply:
I agree with mberz and will switch the order. Check first, setting the data second.